imports/plugins/core/cart/server/no-meteor/util/xformCartGroupToCommonOrder.js

```diff
@@ -13,27 +14,49 @@ export default async function xformCartGroupToCommonOrder(cart, group, context)
   let items = group.itemIds.map((itemId) => cart.items.find((item) => item._id === itemId));
   items = items.filter((item) => !!item); // remove nulls
 
+  // Get productId's from items, and then use ID's to get full Catalog product
+  // to provide missing pricing information
+  const catalogItemIds = items.map((item) => item.productId);
+  const catalogItemsInGroup = await collections.Catalog.find({ "product.productId": { $in: catalogItemIds } }).toArray();
+
   // We also need to add `subtotal` on each item, based on the current price of that item in
   // the catalog. `getFulfillmentGroupTaxes` uses subtotal prop to calculate the tax.
   // ** If you add any data here, be sure to add the same data to the matching xformOrderGroupToCommonOrder xform
-  items = items.map((item) => ({
-    _id: item._id,
-    attributes: item.attributes,
-    isTaxable: item.isTaxable,
-    parcel: item.parcel,
-    price: item.price,
-    productId: item.productId,
-    productVendor: item.productVendor,
-    quantity: item.quantity,
-    shopId: item.shopId,
-    subtotal: {
-      amount: +toFixed(item.price.amount * item.quantity, 3),
-      currencyCode
-    },
-    taxCode: item.taxCode,
-    title: item.title,
-    variantId: item.variantId,
-    variantTitle: item.variantTitle
+  items = await Promise.all(items.map(async (item) => {
+    let itemPrice = item.price;
+
+    // Get the catalog version of the item to get pricing data from it
+    if (catalogItemsInGroup) {
+      const catalogProduct = catalogItemsInGroup.find((catalogItem) => catalogItem.product.productId === item.productId);
+      if (catalogProduct) {
+        const catalogVariant = findVariantInCatalogProduct(catalogProduct.product, item.variantId);
```

Comment:
The variant returned may be `null`. I think this should throw an error if that happens because it should never happen.